tests/Unit/NumericalAlgorithms/Spectral/Test_Clenshaw.cpp

```diff
@@ -0,0 +1,133 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <string>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "NumericalAlgorithms/Spectral/Clenshaw.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+
+namespace Spectral {
+namespace {
+template <typename BasisFunction, typename DataType>
+DataType evaluate_with_function_basis(const std::vector<double>& coefficients,
+                                      const DataType& x,
+                                      const BasisFunction& f_of_k_and_x) {
+  size_t counter = 0;
+  return std::accumulate(
+      coefficients.begin(), coefficients.end(),
+      make_with_value<DataType>(x, 0.0),
+      [&x, &f_of_k_and_x, &counter](auto& sum, const auto& coefficient) {
+        counter++;
+        return sum + coefficient * f_of_k_and_x(counter - 1, x);
+      });
+}
+// Check that for small arguments the clenshaw expansion and the direct cosine
+// evaluation agree exactly.
```

Comment:
The don't agree _exactly_ because of floating point.  I'd just remove the word "exactly".  I don't think further explanation is needed, since anyone looking at an algorithm like this will understand that sort of thing.